Project-Descriptions-and-Plans/LLVM-Toolchain/llvm-toolchain-pl.md

```diff
@@ -0,0 +1,314 @@
+
+- to incorporate bug fixes from the latest versions of the code;
+- to incorporate new optimizations from the latest versions of the code; and
+- to provide new features only available in the latest versions of the code.
+
+The last of these is particularly important given that C and C++ standards are regularly updated and adding important new features in such updates.
+
+## Who would make use of OpenHW output
+
+Any software developer targeting CORE-V processors and wishing to take advantage of CORE-V specific features will need to use either the CORE-V GCC or CORE-V Clang/LLVM compiler tool chain.
+
+## Summary of Timeline
+<!-- High level view of timeline, for example timeframe for each component -->
+Compiler development is an ongoing process, as the upstream project adopts new features, new language standards, bug fixes and optimizations.  Overall the CORE-V project will mirror the upstream 6-monthly release cycle.
+
+Detailed timelines will be associated with specific Plan Approved (PA) proposals.
+
+## Explanation of why OpenHW should do this project
+<!--- What is the impact of doing/not doing this project on the OpenHW ecosystem. Why is OpenHW best suited to do this project -->
+
+A processor is only useful with a robust, up-to-date, proven compiler tool chain.  CORE-V, being a strict RISC-V derivative, can use a standard RISC-V compiler tool chain, but in these circumstances will not be able to take advantage of any of the added features of the CORE-V processors.  This project will provide a baseline tool chain allowing these features to be used (albeit without optimization) and simulated from the source code level.
+
+Furthermore, the licensing scheme of LLVM (Apache license) makes it particularly appealing to industrial players.  Because it does not require the disclosure of compiler source code, chip manufacturers and system integrators can better protect their differentiating Intellectual Property assets.  This licensing scheme also creates additional opportunities for software companies which can provide high added-value services in the area of compiler tuning for hardware/system-centric companies.
+
+In work done a few years ago, code generated for Embench tests by LLVM 11.0 was 4% quicker that GCC 10.2 code. This is an additional reason to offer LLVM support for CORE-V processors.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+
+The "original" compiler used for the development of RISC-V cores is GCC, licensed under the GNU General Public License, requiring vendors to make source code available.  LLVM offers a "permissive" licensing scheme which some customers find more attractive, since it does not require the source to be made available. As a result LLVM-based compiler chains appear to be at the basis (in whole or in part) of a number of proprietary compilers.
```

Comment:
These sections are really informative and helpful

Reply:
Thank you!